Create desmos query client

ivanng-oursky opened the pull request:
## Changes
- Add desmos query client to client provider
- Use the client in dummy page to query oursky's profile

## Screens

On dummy page

<img width="1024" alt="image" src="https://user-images.githubusercontent.com/106721852/173301562-e7815153-a7ee-45d8-9bf5-4194b518f6c3.png">


refs oursky/likedao#133

hochiw commented on react-app/package.json:
```diff
@@ -4,8 +4,9 @@
-    "@cosmjs/stargate": "^0.28.4",
+    "@cosmjs/proto-signing": "^0.27.1",
+    "@cosmjs/stargate": "^0.27.1",
+    "@desmoslabs/desmjs": "^2.3.1-beta.0",
```
have you tried using desmjs `^3.0.0-alpha.2`? that seems to be the latest and working version for me

ivanng-oursky replied:
Yes, but `desmjs` `v3.0.0` uses `desmos` `v3.0.0`, but on mainnet it seems `desmos` is still on `v2.3.1`

See https://github.com/desmos-labs/desmjs/tree/main#compatibility-table

hochiw replied:
oh sad

hochiw commented on react-app/src/providers/QueryClientProvider.tsx:
```diff
@@ -37,9 +48,19 @@ const QueryClientProvider: React.FC<QueryClientProviderProps> = (props) => {
   return (
     <QueryClientContext.Provider value={value}>
-      {!!queryClient && children}
+      {!!(queryClient && desmosQueryClient) && children}
```
can you help to change this whole thing to `!= null`? 🙏

ivanng-oursky replied:
okay changed, do you mean like this?

```ts
{(queryClient && desmosQueryClient) != null && children}
```

hochiw commented on react-app/tsconfig.json:
```diff
@@ -18,7 +18,8 @@
     "noEmit": true,
-    "jsx": "react-jsx"
+    "jsx": "react-jsx",
+    "downlevelIteration": true
```
what is this for?

ivanng-oursky replied:
Remove, I forgot to remove it

ivanng-oursky commented:
@hochiw I cleaned up commit history, note the following changes since you last review:

- updated to `queryClient != null && desmosQueryClient != null && children` per requested change from review
- `@desmoslabs/demjs` have been bump to `v2.0.0-beta.5` (2.0.0 > 2.3.1, idk why)
- resolved all `**/@cosmjs/*` to `v0.28.1` so it will stop complaining that we are using a `cosmjs` version that does not exist (some have `v0.28.6` as deps for some reason, but the latest is `v0.28.4`)

Note, I want to use `@desmoslabs/desmjs` `v2.0.0-beta.5`, which has `@cosmjs/*` `v0.28.4` as dependency, but there is some issue with `v2.0.0-beta.5`, see https://github.com/desmos-labs/desmjs/issues/25

hochiw commented:
> @hochiw I cleaned up commit history, note the following changes since you last review:
> 
>     * updated to `queryClient != null && desmosQueryClient != null && children` per requested change from review
> 
>     * `@desmoslabs/demjs` have been bump to `v2.0.0-beta.5` (2.0.0 > 2.3.1, idk why)
> 
>     * resolved all `**/@cosmjs/*` to `v0.28.1` so it will stop complaining that we are using a `cosmjs` version that does not exist (some have `v0.28.6` as deps for some reason, but the latest is `v0.28.4`)
> 
> 
> Note, I want to use `@desmoslabs/desmjs` `v2.0.0-beta.5`, which has `@cosmjs/*` `v0.28.4` as dependency, but there is some issue with `v2.0.0-beta.5`, see [desmos-labs/desmjs#25](https://github.com/desmos-labs/desmjs/issues/25)

seems can install with `v2.0.0-beta.4` with dependencies set to `v0.28.4`. Looks like `beta.4` -> `beta.5` is some dependency update with improvement to the amino types, since we are not touching that part please see if it's functional with `v0.28.4` set as the version for the dependencies 🙏

hochiw commented:
also i see that you are putting package installing, config modification, client implementation and integration in a single commit, probably should split them into multiple commits next time so that they are easier to understand

ivanng-oursky commented:
> seems can install with v2.0.0-beta.4 with dependencies set to v0.28.4. Looks like beta.4 -> beta.5 is some dependency update with improvement to the amino types, since we are not touching that part please see if it's functional with `v0.28.4` set as the version for the dependencies 🙏

Yes upgrading to v0.28.4 works, I think partly due to specifying dependency resolutions in `package.json` for all cosmjs modules, desmjs is also using v0.28.4 too. 



> also i see that you are putting package installing, config modification, client implementation and integration in a single commit, probably should split them into multiple commits next time so that they are easier to understand

My bad sorry, I will do that next time.

hochiw commented on react-app/package.json:
```diff
@@ -64,6 +72,17 @@
     "stylelint-scss": "^4.2.0",
     "tailwindcss": "^3.0.24"
   },
+  "resolutions": {
+    "**/@cosmjs/amino": "0.28.4",
+    "**/@cosmjs/crypto": "0.28.4",
+    "**/@cosmjs/encoding": "0.28.4",
+    "**/@cosmjs/math": "0.28.4",
+    "**/@cosmjs/proto-signing": "0.28.4",
+    "**/@cosmjs/stargate": "0.28.4",
+    "**/@cosmjs/stream": "0.28.4",
+    "**/@cosmjs/tendermint-rpc": "0.28.4",
+    "**/@cosmjs/utils": "0.28.4"
+  },
```
do we still need resolutions? `0.28.4` should be pass `^0.28.1`?

ivanng-oursky replied:
Seems no, removed 🙇‍♂️

hochiw commented on react-app/package.json:
```diff
@@ -4,15 +4,15 @@
+    "@cosmjs/stargate": "^0.28.1",
+    "@cosmjs/stream": "^0.28.1",
+    "@cosmjs/tendermint-rpc": "^0.28.1",
+    "@cosmjs/utils": "^0.28.1",
```
oh sorry that's not what I meant, we specify `^0.28.4` but since `0.28.4` is part of `^0.28.1` so we don't need resolutions for dependencies with `^0.28.1` to use `0.28.4`

ivanng-oursky replied:
oh I get now, I've updated
